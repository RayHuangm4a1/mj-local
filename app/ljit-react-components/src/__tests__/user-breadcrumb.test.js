import React from 'react';
import { shallow, mount, } from 'enzyme';
import UserBreadcrumb from '../components/user-breadcrumb';

describe('UserBreadcrumb', () => {
	it('handle default props', () => {
		const {
			data,
			targetKey,
			omitText,
			className,
			separator,
			onClickUser,
			isShowingCurrentCount,
		} = UserBreadcrumb.defaultProps;

		expect(data).toEqual([]);
		expect(targetKey).toEqual('');
		expect(omitText).toEqual('......',);
		expect(className).toEqual('',);
		expect(separator).toEqual('>',);
		expect(onClickUser).toBeInstanceOf(Function);
		expect(isShowingCurrentCount).toBe(true);
	});

	it('should render correctly', () => {
		const data = ['mock-data',];
		const wrapper = shallow(
			<UserBreadcrumb
				data={data}
				targetKey=""
				omitText="......"
				className="mock-className"
				separator=">"
				onClickUser={() => {}}
				isShowingCurrentCount={true}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-user-breadcrumb', () => {
		const wrapper = shallow(<UserBreadcrumb />);

		expect(wrapper.hasClass('ljit-user-breadcrumb')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<UserBreadcrumb className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});
	it('should handle onClickUser', () => {
		const mockOnClickUser = jest.fn();
		const wrapper = shallow(
			<UserBreadcrumb
				data={['mock-data', 'mock-data-two',]}
				onClickUser={mockOnClickUser}
			/>
		);
		const ActiveUserSpan = wrapper.find('.ljit-user-breadcrumb__crumb--active').at(0);

		ActiveUserSpan.simulate('click');
		expect(mockOnClickUser).toHaveBeenCalledTimes(1);
		expect(mockOnClickUser).toHaveBeenCalledWith('mock-data');
	});
	it('should mount in a full DOM', () => {
		const data = ['mock-data',];
		const className = 'mock-class';
		const targetKey = '';
		const omitText = '......';
		const separator = '>';
		const onClickUser = () => {};
		const isShowingCurrentCount = true;

		const wrapper = mount(
			<UserBreadcrumb
				data={data}
				targetKey={targetKey}
				omitText={omitText}
				className={className}
				separator={separator}
				onClickUser={onClickUser}
				isShowingCurrentCount={isShowingCurrentCount}
			/>
		);

		expect(wrapper.props().data).toBe(data);
		expect(wrapper.props().targetKey).toBe(targetKey);
		expect(wrapper.props().omitText).toBe(omitText);
		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().separator).toBe(separator);
		expect(wrapper.props().onClickUser).toEqual(onClickUser);
		expect(wrapper.props().isShowingCurrentCount).toEqual(isShowingCurrentCount);
	});
});
