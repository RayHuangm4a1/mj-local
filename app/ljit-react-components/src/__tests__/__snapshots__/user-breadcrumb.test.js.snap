// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBreadcrumb should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserBreadcrumb
    className="mock-className"
    data={
      Array [
        "mock-data",
      ]
    }
    isShowingCurrentCount={true}
    omitText="......"
    onClickUser={[Function]}
    separator=">"
    targetKey=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <React.Fragment>
          <span
            className="ljit-user-breadcrumb__crumb"
          >
            mock-data
          </span>
          <span
            className="ljit-user-breadcrumb__current-count"
          >
            (第1层)
          </span>
        </React.Fragment>,
      ],
      "className": "ljit-user-breadcrumb mock-className",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "crumb-0",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <span
              className="ljit-user-breadcrumb__crumb"
            >
              mock-data
            </span>,
            <span
              className="ljit-user-breadcrumb__current-count"
            >
              (第1层)
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mock-data",
              "className": "ljit-user-breadcrumb__crumb",
            },
            "ref": null,
            "rendered": "mock-data",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "(第1层)",
              "className": "ljit-user-breadcrumb__current-count",
            },
            "ref": null,
            "rendered": "(第1层)",
            "type": "span",
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <React.Fragment>
            <span
              className="ljit-user-breadcrumb__crumb"
            >
              mock-data
            </span>
            <span
              className="ljit-user-breadcrumb__current-count"
            >
              (第1层)
            </span>
          </React.Fragment>,
        ],
        "className": "ljit-user-breadcrumb mock-className",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "crumb-0",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span
                className="ljit-user-breadcrumb__crumb"
              >
                mock-data
              </span>,
              <span
                className="ljit-user-breadcrumb__current-count"
              >
                (第1层)
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mock-data",
                "className": "ljit-user-breadcrumb__crumb",
              },
              "ref": null,
              "rendered": "mock-data",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(第1层)",
                "className": "ljit-user-breadcrumb__current-count",
              },
              "ref": null,
              "rendered": "(第1层)",
              "type": "span",
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
